proposals/0000-unicode-for-string-processing.md

```diff
@@ -0,0 +1,1321 @@
+| `Bidi_Control`                      | single-scalar                 |                                   |
+| `Bidi_Mirrored`                     | first-scalar                  |                                   |
+| <br> **Miscellaneous**              |                               |                                   |
+| `Math`                              | first-scalar                  |                                   |
+| `Quotation_Mark`                    | first-scalar                  |                                   |
+| `Dash`                              | first-scalar                  |                                   |
+| `Sentence_Terminal`                 | first-scalar                  |                                   |
+| `Terminal_Punctuation`              | first-scalar                  |                                   |
+| `Diacritic`                         | single-scalar                 |                                   |
+| `Extender`                          | single-scalar                 |                                   |
+| `Grapheme_Base`                     | single-scalar                 |                                   |
+| `Grapheme_Extend`                   | single-scalar                 |                                   |
+
+</details>
+
+To invert a Unicode property character class, use `\P{...}`.
+
+When using `RegexBuilder` syntax, Unicode property classes are available through the following methods on  `CharacterClass`:
+
+- `static func generalCategory(_: Unicode.GeneralCategory) -> CharacterClass`
+- `static func binaryProperty(_: KeyPath<UnicodeScalar.Properties, Bool>, value: Bool = true) -> CharacterClass`
+- `static func named(_: String) -> CharacterClass`
+- `static func age(_: Unicode.Version) -> CharacterClass`
+- `static func numericType(_: Unicode.NumericType) -> CharacterClass`
+- `static func numericValue(_: Double) -> CharacterClass`
+- `static func lowercaseMapping(_: String) -> CharacterClass`
+- `static func uppercaseMapping(_: String) -> CharacterClass`
+- `static func titlecaseMapping(_: String) -> CharacterClass`
+- `static func canonicalCombiningClass(_: Unicode.CanonicalCombiningClass) -> CharacterClass`
+
+You can see the full `CharacterClass` API with documentation comments in the **Custom Classes** section, below.
+
+#### POSIX character classes: `[:NAME:]` or `\p{NAME}`
+
+**POSIX character classes** represent concepts that we'd like to define at all semantic levels. We propose the following definitions, some of which have been described above. When matching with grapheme cluster semantics, Unicode properties are extended to `Character`s as described in the rationale above, and as shown in the table below. That is, for POSIX class `[:word:]`, any `Character` that starts with a matching scalar is a match, while for `[:digit:]`, a matching `Character` must only comprise a single Unicode scalar value.
```

Comment:
This sounds like it's contrasting with the above. Is `[:word:]` the same as `\w` or not?